packages/subgraph/schema.graphql

```diff
@@ -107,70 +107,102 @@ type Dao @entity {
   balances: [Balance!] @derivedFrom(field: "dao")
   contractPermissionIds: [ContractPermissionId!]! @derivedFrom(field: "dao")
   permissions: [Permission!]! @derivedFrom(field: "dao")
-  plugins: [DaoPlugin!] @derivedFrom(field: "dao")
+  plugins: [PluginInstallation!]! @derivedFrom(field: "dao")
   proposals: [Proposal!] @derivedFrom(field: "dao")
-  "TODO: remove or rename to `plugins` once plugins are seperated from the core"
-  corePlugins: [Plugin!] @derivedFrom(field: "dao")
   trustedForwarder: Bytes
   signatureValidator: Bytes
   standardCallbacks: [StandardCallback!] @derivedFrom(field: "dao")
 }
 
 # Plugins
 
+type PluginRepo @entity(immutable: true) {
+  id: ID! # address
+  subdomain: String!
+  versions: [PluginVersion!]! @derivedFrom(field: "pluginRepo")
+  preparations: [PluginPreparation!]! @derivedFrom(field: "pluginRepo")
+  installations: [PluginInstallation!] @derivedFrom(field: "appliedPluginRepo")
+}
+
 type PluginSetup @entity(immutable: true) {
   id: ID! # contract address
   versions: [PluginVersion!]! @derivedFrom(field: "pluginSetup")
-  installations: [Plugin!]! @derivedFrom(field: "pluginSetup")
 }
 
 type PluginVersion @entity(immutable: true) {
-  id: ID! # pluginRepo + version index
+  id: ID! # pluginRepo + release + build
   pluginRepo: PluginRepo!
-  semanticVersion: [BigInt!]!
-  pluginSetup: PluginSetup!
-  contentURI: Bytes!
+  pluginSetup: PluginSetup
+  release: Int!
+  build: Int!
+  metadata: Bytes!
+  preparations: [PluginPreparation!]! @derivedFrom(field: "pluginVersion")
+  installations: [PluginInstallation!] @derivedFrom(field: "appliedVersion")
+}
+
+type PluginPermission @entity(immutable: true) {
+  id: ID! # PluginPreparation + operation + where + who + permissionId
+  pluginPreparation: PluginPreparation!
+  operation: PermissionOperation!
+  where: Bytes!
+  who: Bytes!
+  condition: Bytes
+  permissionId: Bytes!
 }
 
-type PluginRepo @entity(immutable: true) {
-  id: ID! # address
-  subdomain: String!
-  versions: [PluginVersion!]! @derivedFrom(field: "pluginRepo")
+enum PermissionOperation {
+  Grant
+  Revoke
+  GrantWithCondition
 }
 
-type PluginHelper @entity(immutable: true) {
-  id: ID! # address
-  plugin: Plugin!
+type PluginPreparation @entity(immutable: true) {
+  id: ID! # psp setupId + installationId
+  creator: Bytes!
+  dao: Dao!
+  setupId: Bytes! # the psp setupId
```

Comment:
Shouldn't this be named `preparedSetupId` (since this is how we name it in the PSP)?

Reply:
done